pkg/controller/operators/operator_controller_test.go

```diff
@@ -204,6 +204,44 @@ var _ = Describe("Operator Controller", func() {
 				})
 			})
 
+			Context("when multiple types of a gvk are labeled", func() {
+				expectedServiceAccountOrder := []string{"sa-1", "sa-2", "sa-3", "sa-4", "sa-5", "sa-6", "sa-7", "sa-8", "sa-9"}
+				BeforeEach(func() {
+					newObjs := make([]runtime.Object, 10)
+					for i, saName := range expectedServiceAccountOrder {
+						// Create objects in reverse order to ensure they are eventually ordered
+						newObjs[9-i] = testobj.WithNamespacedName(
+							&types.NamespacedName{Namespace: namespace, Name: saName}, &corev1.ServiceAccount{},
+						)
+					}
+
+					for _, obj := range newObjs {
+						Expect(k8sClient.Create(ctx, obj.(client.Object))).To(Succeed())
+					}
+
+					objs = append(objs, newObjs...)
+					expectedRefs = append(expectedRefs, toRefs(scheme, newObjs...)...)
+				})
+
+				It("should list each of the component references in alphabetical order by namespace and name", func() {
+					Eventually(func() ([]operatorsv1.RichReference, error) {
+						err := k8sClient.Get(ctx, name, operator)
+						return operator.Status.Components.Refs, err
+					}, timeout, interval).Should(ConsistOf(expectedRefs))
```

Comment:
Should we split this out into it's own individual spec?

Reply:
Sorry, it's own individual spec?

Reply:
Err, s/"individual spec"/an "It" container if that makes more sense?

Basically, move this check into an "It" container:

```go
It("should eventually contain a status.Components.Refs that match the expected ref count", func() {
	Eventually(func() ([]operatorsv1.RichReference, error) {
		err := k8sClient.Get(ctx, name, operator)
		return operator.Status.Components.Refs, err
	}, timeout, interval).Should(ConsistOf(expectedRefs))
})
...
```

But I don't think this is super important given the nature of this bug fix.